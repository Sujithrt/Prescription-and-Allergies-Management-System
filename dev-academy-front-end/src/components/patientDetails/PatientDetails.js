import React from 'react';
import {
    CalendarToday,
    LocationSearching,
    MailOutline,
    PermIdentity,
} from "@material-ui/icons";
import ContactPhoneOutlinedIcon from '@mui/icons-material/ContactPhoneOutlined';

/**
 * Displays Patients Details to the user
 * @param {patient} props 
 * @returns 
 */
export default function PatientDetails(props) {
    return (
        <>
            <div className="userShowUsername">Registration Number: {props.patient.id}</div>
            <span className="userShowTitle">Patient Details</span>
            <div className="patientDetails">
                <PermIdentity className="userShowIcon" />
                <span className="userShowInfoTitle">{props.patient.firstName} {props.patient.lastName}</span>
            </div>
            <div className="patientDetails">
                <CalendarToday className="userShowIcon" />
                <span className="userShowInfoTitle">{props.patient.age}</span>
            </div>
            <span className="userShowTitle">Contact Details</span>
            <div className="patientDetails">
                <MailOutline className="userShowIcon" />
                <span className="userShowInfoTitle">{props.patient.email}</span>
            </div>
            <div className="patientDetails">
                <ContactPhoneOutlinedIcon className="userShowIcon" />
                <span className="userShowInfoTitle">{props.patient.phoneNumber}</span>
            </div>
            <div className="patientDetails">
                <LocationSearching className="userShowIcon" />
                <span className="userShowInfoTitle">{props.patient.city} | {props.patient.state}</span>
            </div>
        </>
    )
}
