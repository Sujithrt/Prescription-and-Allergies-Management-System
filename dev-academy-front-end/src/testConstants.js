export const patient = {
    "id": 1001,
    "firstName": "Leanne",
    "lastName": "Graham",
    "gender": "Male",
    "age": 21,
    "phoneNumber": 9123456780,
    "email": "leanne.graham@gmail.com",
    "city": "Bangalore",
    "state": "Karnataka",
    "medicalConditions": [
        {
            "id": 1,
            "name": "Cephalalgia (headache)"
        },
        {
            "id": 3,
            "name": "Otitis externa (swimmer's ear)"
        },
        {
            "id": 9,
            "name": "Contusion (bruise)"
        }
    ],
    "allergies": [
        {
            "id": 7,
            "name": "Aspirin Allergy",
            "reaction": "Tiredness",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 4,
                "name": "Aspirin"
            },
            "encounter": {
                "id": 10,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 8,
            "name": "Paracetamol",
            "reaction": "sdfd",
            "severity": "MILD",
            "status": "ACTIVE",
            "ingredient": {
                "id": 1,
                "name": "Paracetamol"
            },
            "encounter": {
                "id": 13,
                "date": "2021-10-26",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 13,
            "name": "Ibuprofin Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 20,
                "date": "2021-10-29",
                "patient": {
                    "id": 1001,
                    "firstName": "Leanne",
                    "lastName": "Graham",
                    "gender": "Male",
                    "age": 21,
                    "phoneNumber": 9123456780,
                    "email": "leanne.graham@gmail.com",
                    "city": "Bangalore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 1,
                            "name": "Cephalalgia (headache)"
                        },
                        {
                            "id": 3,
                            "name": "Otitis externa (swimmer's ear)"
                        },
                        {
                            "id": 9,
                            "name": "Contusion (bruise)"
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        },
        {
            "id": 14,
            "name": "Paracetamol Allergy",
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
            "ingredient": {
                "id": 3,
                "name": "Ibuprofen"
            },
            "encounter": {
                "id": 1,
                "date": "2021-10-26",
                "patient": {
                    "id": 1003,
                    "firstName": "Clementine",
                    "lastName": "Bauch",
                    "gender": "Female",
                    "age": 56,
                    "phoneNumber": 9123456782,
                    "email": "clementine.bauch@gmail.com",
                    "city": "Mysore",
                    "state": "Karnataka",
                    "medicalConditions": [
                        {
                            "id": 23,
                            "name": "Gestation (pregnancy) "
                        }
                    ]
                },
                "user": {
                    "id": 1,
                    "username": "John",
                    "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                    "email": "john.cage@gmail.com",
                    "enabled": true,
                    "role": "Physician",
                    "authorities": [
                        {
                            "id": 1,
                            "roleCode": "USER",
                            "roleDescription": "User role",
                            "authority": "USER"
                        },
                        {
                            "id": 2,
                            "roleCode": "ADMIN",
                            "roleDescription": "Admin role",
                            "authority": "ADMIN"
                        }
                    ],
                    "credentialsNonExpired": true,
                    "accountNonExpired": true,
                    "accountNonLocked": true
                }
            },
            "patient": {
                "id": 1001,
                "firstName": "Leanne",
                "lastName": "Graham",
                "gender": "Male",
                "age": 21,
                "phoneNumber": 9123456780,
                "email": "leanne.graham@gmail.com",
                "city": "Bangalore",
                "state": "Karnataka",
                "medicalConditions": [
                    {
                        "id": 1,
                        "name": "Cephalalgia (headache)"
                    },
                    {
                        "id": 3,
                        "name": "Otitis externa (swimmer's ear)"
                    },
                    {
                        "id": 9,
                        "name": "Contusion (bruise)"
                    }
                ]
            }
        }
    ]
};

export const user = {
    "id": 1,
    "username": "John",
    "email": "john.cage@gmail.com",
    "role": "Physician",
    "firstName": "John",
    "lastName": "Cage",
    "speciality": "Cardiologist",
    "city": "Bangalore",
    "state": "Karnataka",
    "dob": "1975-06-12T18:30:00.000+00:00",
    "authorities": [
        {
            "id": 1,
            "roleCode": "USER",
            "roleDescription": "User role",
            "authority": "USER"
        },
        {
            "id": 2,
            "roleCode": "ADMIN",
            "roleDescription": "Admin role",
            "authority": "ADMIN"
        }
    ]
};

export const prescription = [
    {
        "id": {
            "encounterId": 1,
            "medicineId": 1
        },
        "encounter": {
            "id": 29,
            "date": "2021-11-07",
            "patient": {
                "id": 1009,
                "firstName": "Glenna",
                "lastName": "Reichert",
                "gender": "Female",
                "age": 65,
                "phoneNumber": 9123456788,
                "email": "glenna.reichert@gmail.com",
                "city": "Kolkata",
                "state": "West Bengal",
                "medicalConditions": [
                    {
                        "id": 6,
                        "name": "Tussis (cough)"
                    },
                    {
                        "id": 7,
                        "name": "Pyrexia (fever)"
                    }
                ]
            },
            "user": {
                "id": 1,
                "username": "John",
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "email": "john.cage@gmail.com",
                "enabled": true,
                "role": "Physician",
                "authorities": [
                    {
                        "id": 1,
                        "roleCode": "USER",
                        "roleDescription": "User role",
                        "authority": "USER"
                    },
                    {
                        "id": 2,
                        "roleCode": "ADMIN",
                        "roleDescription": "Admin role",
                        "authority": "ADMIN"
                    }
                ],
                "credentialsNonExpired": true,
                "accountNonExpired": true,
                "accountNonLocked": true
            }
        },
        "medicine": {
            "id": 1,
            "genericName": "Paracetamol",
            "brandName": "Dolo",
            "category": {
                "id": 1,
                "name": "Antipyretics"
            },
            "ingredients": [
                {
                    "id": 1,
                    "name": "Paracetamol"
                }
            ]
        },
        "dosage": "1 time a day"
    }
];

export const allergy = {
    "id": 19,
    "name": "Paracetamol Allergy",
    "reaction": "Tiredness",
    "severity": "MILD",
    "status": "ACTIVE",
    "ingredient": {
        "id": 1,
        "name": "Paracetamol"
    },
    "encounter": {
        "id": 28,
        "date": "2021-11-07",
        "patient": {
            "id": 1009,
            "firstName": "Glenna",
            "lastName": "Reichert",
            "gender": "Female",
            "age": 65,
            "phoneNumber": 9123456788,
            "email": "glenna.reichert@gmail.com",
            "city": "Kolkata",
            "state": "West Bengal",
            "medicalConditions": [
                {
                    "id": 6,
                    "name": "Tussis (cough)"
                },
                {
                    "id": 7,
                    "name": "Pyrexia (fever)"
                }
            ]
        },
        "user": {
            "id": 1,
            "username": "John",
            "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
            "email": "john.cage@gmail.com",
            "enabled": true,
            "role": "Physician",
            "authorities": [
                {
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                    "authority": "USER"
                },
                {
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                    "authority": "ADMIN"
                }
            ],
            "credentialsNonExpired": true,
            "accountNonExpired": true,
            "accountNonLocked": true
        }
    },
    "patient": {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "gender": "Female",
        "age": 65,
        "phoneNumber": 9123456788,
        "email": "glenna.reichert@gmail.com",
        "city": "Kolkata",
        "state": "West Bengal",
        "medicalConditions": [
            {
                "id": 6,
                "name": "Tussis (cough)"
            },
            {
                "id": 7,
                "name": "Pyrexia (fever)"
            }
        ]
    }
};

export const recentPatients = [
    {
        "id": 1009,
        "firstName": "Glenna",
        "lastName": "Reichert",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-09"
    },
    {
        "id": 1008,
        "firstName": "Nicholas",
        "lastName": "Runolfsdottir",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1021,
        "firstName": "Leroy",
        "lastName": "Blair",
        "age": 21,
        "gender": "Male",
        "date": "2021-11-07"
    },
    {
        "id": 1027,
        "firstName": "Alanna",
        "lastName": "Cantrell",
        "age": 65,
        "gender": "Female",
        "date": "2021-11-07"
    },
    {
        "id": 1016,
        "firstName": "Osbert",
        "lastName": "Valdez",
        "age": 32,
        "gender": "Male",
        "date": "2021-11-02"
    },
    {
        "id": 1024,
        "firstName": "Emmett",
        "lastName": "Marshall",
        "age": 45,
        "gender": "Male",
        "date": "2021-11-02"
    }
];

export const prescriptionData = [
    {
        "brandName": "Tylenol",
        "genericName": "Acetaminophen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Advil",
        "genericName": "Ibuprofen",
        "dosage": "3 times a day",
        "date": "2021-10-26"
    },
    {
        "brandName": "Dolo",
        "genericName": "Paracetamol",
        "dosage": "1 time a day",
        "date": "2021-10-26"
    }
];

export const medicineList = [
    {
        "id": 1,
        "brandName": "Dolo",
        "cause": "Incompatible with patients' medical conditions"
    },
    {
        "id": 13,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 17,
        "brandName": "Vibramycin",
        "cause": ""
    },
    {
        "id": 22,
        "brandName": "Betadine",
        "cause": ""
    },
    {
        "id": 25,
        "brandName": "Iodex",
        "cause": ""
    },
    {
        "id": 46,
        "brandName": "Desyrel",
        "cause": "Incompatible with patients' medical conditions"
    }
];