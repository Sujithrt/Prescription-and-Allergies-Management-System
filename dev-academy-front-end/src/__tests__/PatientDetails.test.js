import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow } from 'enzyme';
import { patient } from './../testConstants'
import PatientDetails from '../components/patientDetails/PatientDetails'

describe("PatientDetails Test", ()=>{
  it("PatientDetails is rendered as expected", ()=>{
    const patientDetails = shallow(<PatientDetails patient={patient}/>);
    expect(shallowToJson(patientDetails)).toMatchSnapshot();
  });
});