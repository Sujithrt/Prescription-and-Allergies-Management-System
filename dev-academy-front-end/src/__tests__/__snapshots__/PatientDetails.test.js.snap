// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientDetails Test PatientDetails is rendered as expected 1`] = `
<Fragment>
  <div
    className="userShowUsername"
  >
    Registration Number: 
    1001
  </div>
  <span
    className="userShowTitle"
  >
    Patient Details
  </span>
  <div
    className="patientDetails"
  >
    <Memo(ForwardRef(PermIdentityIcon))
      className="userShowIcon"
    />
    <span
      className="userShowInfoTitle"
    >
      Leanne
       
      Graham
    </span>
  </div>
  <div
    className="patientDetails"
  >
    <Memo(ForwardRef(CalendarTodayIcon))
      className="userShowIcon"
    />
    <span
      className="userShowInfoTitle"
    >
      21
    </span>
  </div>
  <span
    className="userShowTitle"
  >
    Contact Details
  </span>
  <div
    className="patientDetails"
  >
    <Memo(ForwardRef(MailOutlineIcon))
      className="userShowIcon"
    />
    <span
      className="userShowInfoTitle"
    >
      leanne.graham@gmail.com
    </span>
  </div>
  <div
    className="patientDetails"
  >
    <Memo(ForwardRef(ContactPhoneOutlinedIcon))
      className="userShowIcon"
    />
    <span
      className="userShowInfoTitle"
    >
      9123456780
    </span>
  </div>
  <div
    className="patientDetails"
  >
    <Memo(ForwardRef(LocationSearchingIcon))
      className="userShowIcon"
    />
    <span
      className="userShowInfoTitle"
    >
      Bangalore
       | 
      Karnataka
    </span>
  </div>
</Fragment>
`;
